### DISCUSSION TITLE
[2023-10-16] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8079] [Weekly Demo Day]
The Searcher input is now interpreted using parser; the filtering works better, and also applying suggestion works. It's better than the old GUI because:
1. It adds space to operator if needed (like `2 +` input turns to `2 + operator1` instead of `2 +operator1`)
2. It fills the qualified name segments properly
3. "Entering modules" works as in the newest design: the module name is added to input with a dot, and its content is then displayed.
4. The space is not so aggressively added - instead, we just move the cursor past the character (like, past a comma or closing bracket).

https://github.com/enso-org/enso/assets/3919101/c5ff116c-3fa9-4a98-ba03-cba404e96c82

**Comment by mwu-tow** [comment]
Wrapping a number of remaining loose ends with CI:
* Releasing IDE2 artifacts;
* Fixes for case when Backend tests results were not correctly uploaded;
* Automatic cleaning CI servers;
* Bumping some outdated third-party actions.

GUI2:
Started work on alias analysis (https://github.com/enso-org/enso/issues/7863): had some discussions with the Engine Team regarding the language semantics. I expect some implementation work this week (not necessarily complete though).

**Comment by wdanilo** [comment]
On my end:
- I created next parts of the website. Almost all node-related animations are there. 
- I spent literally days on making the mock data animations and being sure it runs correctly. It's there and works well. 
- I reworked all captions for the website to keep them shorter and easier to read. I also re-worked what we show in animations to keep them short as well.
 
Here is a preview, I will post full video at the end of the day, as there is also a geo-map animation, but I broke it before this call:


https://github.com/enso-org/enso/assets/1623053/d2b9d961-1283-47c0-88df-7b0789e9dbc3



What is missing:
- Connections between nodes.
- Fixing bugs (esp. when resizing screen) and polishing the feel.
- Buy page (small one).

We have a hard deadline of the website the beginning of next week, so when I will release preview this week, I will be thankful for extensive testing.

**Comment by vitvakatu** [comment]
Working on documentation panel implementation.

A lot of work on the backend side of things, implementing intermediate representation for entries documentation and utilities to process doc sections. 
With help from @Frizi, implemented [`ReactiveDb`](https://github.com/enso-org/enso/pull/8052) abstraction, which allows us building arbitrary indexes on top of the Suggestion database.

Also, fixes from the engine team allowed me to use a real suggestion database instead of mocks, which uncovered many visual issues.


https://github.com/enso-org/enso/assets/6566674/5e0f35f5-a734-49c0-9416-f00e633c4e37

**Comment by MichaelMauderer** [comment]
* Finished work on https://github.com/enso-org/enso/issues/7990, review/merge is pending a test of the updated Language Server that should solve some issues with timeouts. Should move forward tomorrow. 
* Started work on #7786 and should finish this week.

**Comment by kazcw** [comment]
- Finished adding document references to Spans #8039 (in CI).
- Merging progress on #7922 to avoid further merge conflicts (applying review).
- Implementing type-driven AST traversal #8025, will finish today.

**Comment by somebody1234** [comment]
- Mostly working on dashboard last week, but:
- Implemented #8075
  - Turns out it was relying on old code, so hover tooltips are currently broke on develop, currently this is planned to be fixed in #8074 (minor GUI2 spring cleaning PR)
- (PSA?) A recent dashboard PR broke `npm run dev`, hotfix incoming: #8078

https://github.com/enso-org/enso/assets/4046547/554c141a-1ae6-4214-8578-2eac2c319937

**Comment by Frizi** [comment]
- Helped with typescript AST representation design, in preparation for allowing mutable AST operations. Cleaned up AST typescript code generator. https://github.com/enso-org/enso/pull/8028
- Helped with reactive database and index implementation needed for more advanced suggestion database features (such as hierarchy traversal) https://github.com/enso-org/enso/pull/8058
- Added finding of the AST subtree matching current execution context stack, and creating nodes from statements defined in it. Still working on "proto-widgets" view that use that AST in each node view. https://github.com/enso-org/enso/issues/7791

**Comment by hubertp** [comment]
Relevant Engine changes:
- Fix initialization issues with new GUI #7985 
- Enso VS Extension https://marketplace.visualstudio.com/items?itemName=Enso.enso4vscode
